datastore/redis.go

```diff
@@ -0,0 +1,49 @@
+package datastore
+
+import (
+	"fmt"
+	"github.com/avigailbr/book_service/connectors"
+)
+
+type RedisCache struct {
+	Client *connectors.RedisClient
+}
+
+func NewRedisCache(addr string) (IActivityCacher, error) {
```

Comment:
That's a common practice in languages such C# and Java, but since [golang is a bit different](https://medium.com/@cep21/what-accept-interfaces-return-structs-means-in-go-2fe879e25ee8) it's worth taking into account